<!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta name="description" content="RNA_seq流程"><meta name="keywords" content="RNA_seq"><meta name="author" content="双语之城"><meta name="copyright" content="双语之城"><title>RNA_seq流程 | 双语之城的博客</title><link rel="shortcut icon" href="/logo.png"><link rel="stylesheet" href="/css/index.css?version=1.6.1"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/font-awesome@latest/css/font-awesome.min.css?version=1.6.1"><link rel="dns-prefetch" href="https://cdn.staticfile.org"><link rel="dns-prefetch" href="https://cdn.bootcss.com"><link rel="dns-prefetch" href="https://creativecommons.org"><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  }
} </script><meta name="generator" content="Hexo 5.4.0"></head><body><canvas class="fireworks"></canvas><i class="fa fa-arrow-right" id="toggle-sidebar" aria-hidden="true"></i><div id="sidebar"><div class="toggle-sidebar-info text-center"><span data-toggle="Toggle article">Toggle site</span><hr></div><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar"></div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#1-RNA-seq%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD"><span class="toc-number">1.</span> <span class="toc-text">1.RNA_seq数据下载</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#2-FASTQC%E8%B4%A8%E6%8E%A7%E5%88%86%E6%9E%90"><span class="toc-number">2.</span> <span class="toc-text">2.FASTQC质控分析</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Trimmomatic%E5%BA%8F%E5%88%97%E4%BF%AE%E5%89%AA"><span class="toc-number">3.</span> <span class="toc-text">Trimmomatic序列修剪</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#TopHat2-cufflinks%E6%B5%81%E7%A8%8B"><span class="toc-number">4.</span> <span class="toc-text">TopHat2+cufflinks流程</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#4-1-tophat2%E5%BA%8F%E5%88%97%E6%AF%94%E5%AF%B9"><span class="toc-number">4.1.</span> <span class="toc-text">4.1 tophat2序列比对</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-2-Cufflinks%E5%BA%8F%E5%88%97%E7%BB%84%E8%A3%85"><span class="toc-number">4.2.</span> <span class="toc-text">4.2 Cufflinks序列组装</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Hisat2-StringTie-Ballgown%E6%B5%81%E7%A8%8B"><span class="toc-number">5.</span> <span class="toc-text">Hisat2+StringTie+Ballgown流程</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#5-1-Hisat2-stringtie"><span class="toc-number">5.1.</span> <span class="toc-text">5.1 Hisat2+stringtie</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5-2-Ballgown"><span class="toc-number">5.2.</span> <span class="toc-text">5.2 Ballgown</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE-%E8%81%9A%E7%B1%BB%E5%88%86%E6%9E%90"><span class="toc-number">6.</span> <span class="toc-text">差异表达+聚类分析</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#6-1-TopHat2-Cufflinks%E6%B5%81%E7%A8%8B"><span class="toc-number">6.1.</span> <span class="toc-text">6.1 TopHat2 + Cufflinks流程</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#6-2-Hisat2-StringTie-Ballgown%E6%B5%81%E7%A8%8B"><span class="toc-number">6.2.</span> <span class="toc-text">6.2 Hisat2+StringTie+Ballgown流程</span></a></li></ol></li></ol></div></div><div class="author-info hide"><div class="author-info__avatar text-center"><img src="/img/avatar.png"></div><div class="author-info__name text-center">双语之城</div><div class="author-info__description text-center"></div><hr><div class="author-info-articles"><a class="author-info-articles__archives article-meta" href="/archives"><span class="pull-left">Articles</span><span class="pull-right">24</span></a><a class="author-info-articles__tags article-meta" href="/tags"><span class="pull-left">Tags</span><span class="pull-right">19</span></a></div></div></div><div id="content-outer"><div id="top-container" style="background-image: url(/sabar.jpg)"><div id="page-header"><span class="pull-left"> <a id="site-name" href="/">双语之城的博客</a></span><i class="fa fa-bars toggle-menu pull-right" aria-hidden="true"></i><span class="pull-right menus"><a class="site-page" href="/">Home</a><a class="site-page" href="/archives">Archives</a><a class="site-page" href="/tags">Tags</a><a class="site-page" href="/categories">Categories</a></span></div><div id="post-info"><div id="post-title">RNA_seq流程</div><div id="post-meta"><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2019-05-17</time></div></div></div><div class="layout" id="content-inner"><article id="post"><div class="article-container" id="post-content"><p>假期姑且总结一下……</p>
<span id="more"></span>

<h1 id="1-RNA-seq数据下载"><a href="#1-RNA-seq数据下载" class="headerlink" title="1.RNA_seq数据下载"></a>1.RNA_seq数据下载</h1><p>在NCBI GEO<br>DataSet中选择GSE107744作为所选数据集。该数据为衰老酵母MEP细胞的野生型（wt）、swd1、spp1、set1CD四种类型分别进行YPD 0h/ 7.5h/ 24h/ 48h。芯片从GSM2877741到GSM2877794共计54张芯片。</p>
<p><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107744">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107744</a></p>
<p>获取每张芯片对应的SRR编号</p>
<img src="/2019/05/17/RNA-seq/831114e30ff9d23befe17fcc99d1e74d.png" class="">

<p>对应到EBI数据库中，获取各SRR对应的ftp</p>
<img src="/2019/05/17/RNA-seq/fb85a6bd4ae423620851a4f7229b1984.png" class="">

<p>下载并配置Aspera的运行环境</p>
<p>Aspera是高速下载的工具，最快可达到100M/s。</p>
<p>进入Aspera官网的Downloads界面，选中aspera connect<br>server，点击Wwindows图标，选择v3.6.2版本，点击Download进行下载。</p>
<p>配置Aspera的运行环境</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">#安装aspera connect</span><br><span class="line">tar zxf aspera-connect-3.5.1.92523-linux-64.tar.gz</span><br><span class="line">$ sh aspera-connect-3.5.1.92523-linux-64.sh</span><br><span class="line">#将aspera运行文件复制到运行路径（或者直接添加环境变量）</span><br><span class="line">cd  #进入根目录</span><br><span class="line">ls -a   #检查是否已安装好.aspera</span><br><span class="line">.aspera/ect/</span><br><span class="line">#进入etc目录</span><br><span class="line">cp -r etc/. /mnt/</span><br><span class="line">#开放端口</span><br><span class="line">#进入root</span><br><span class="line">su</span><br><span class="line">iptables -A INPUT -p udp --dport 33001 -j ACCEPT</span><br><span class="line">iptables -A OUTPUT -p udp --dport 33001 -j ACCEPT</span><br><span class="line">iptables -L -n</span><br><span class="line">exit</span><br><span class="line"></span><br><span class="line">#下载方法（以SRR6350254为例）</span><br><span class="line">ascp -QT -l 500M -P 33001 -i asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR635/004/SRR6350254/SRR6350254.fastq.gz /mnt/sra</span><br></pre></td></tr></table></figure>

<h1 id="2-FASTQC质控分析"><a href="#2-FASTQC质控分析" class="headerlink" title="2.FASTQC质控分析"></a>2.FASTQC质控分析</h1><figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">#利用fastqc分析所有质控</span><br><span class="line">fastqc new_SRR6350254.fastq</span><br><span class="line">#用multiqc工具总结</span><br><span class="line">bash Anaconda2-5.3.1-Linux-x86_64.sh</span><br><span class="line">conda install -c bioconda multiqc</span><br><span class="line">multiqc --help</span><br><span class="line">multiqc .       #在包含所有fastqc.zip的文件夹中运行</span><br></pre></td></tr></table></figure>


<img src="/2019/05/17/RNA-seq/f22ae8f427c8107ac4c6f13d89f04b84.png" class="">

<img src="/2019/05/17/RNA-seq/6665c1cae86ae1dff49329aa43178678.png" class="">

<p>图 1 序列质量分布 图 2 序列质量打分</p>
<img src="/2019/05/17/RNA-seq/4091fee285a6f6c85eac38998adf93fb.png" class="">

<img src="/2019/05/17/RNA-seq/157ba983978255955454d11aedf68ab0.png" class="">

<p>图 3 GC值情况 图 4 N值情况</p>
<img src="/2019/05/17/RNA-seq/22432529ecd256f9f4aa5962eac37165.png" class="">

<img src="/2019/05/17/RNA-seq/c1825a0e248d0d2dd21367155ebfe00b.png" class="">

<p>图 5 序列重复水平 图 6 过表达情况</p>
<ol start="3">
<li><h1 id="Trimmomatic序列修剪"><a href="#Trimmomatic序列修剪" class="headerlink" title="Trimmomatic序列修剪"></a>Trimmomatic序列修剪</h1><figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">for i in `seq 197 254`</span><br><span class="line">do   </span><br><span class="line">java -jar ./trimmomatic-0.39.jar SE -threads 6 -phred33 -trimlog SRR6350$&#123;i&#125;_trimmomatic.log SRR6350$&#123;i&#125;.fastq new_SRR6350$&#123;i&#125;.fastq HEADCROP:11</span><br><span class="line">done</span><br></pre></td></tr></table></figure>
<p>运行Trimmomatic bash trimmomatic.sh</p>
</li>
<li><h1 id="TopHat2-cufflinks流程"><a href="#TopHat2-cufflinks流程" class="headerlink" title="TopHat2+cufflinks流程"></a>TopHat2+cufflinks流程</h1></li>
</ol>
<h2 id="4-1-tophat2序列比对"><a href="#4-1-tophat2序列比对" class="headerlink" title="4.1 tophat2序列比对"></a>4.1 tophat2序列比对</h2><p>下载tophat2，解压，将文件软连接到环境变量中</p>
<p>sudo ln -s /mnt/g/RNA_seq/tophat-2.1.1.Linux_x86_64/tophat2<br>/usr/local/bin/tophat2</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">for i in `seq 197 254`</span><br><span class="line">do   </span><br><span class="line">tophat2 -p 4 -G /mnt/g/RNA_seq/sra/tophat2_align/Saccharomyces_cerevisiae.R64-1-1.86.gtf -o tophat2_align/new_SRR6350$&#123;i&#125;_thout_new/ /mnt/g/RNA_seq/sra/bowtie2_align/Bowtie2Index/genome /mnt/g/RNA_seq/sra/new_SRR6350$&#123;i&#125;.fastq &gt;new_SRR6350$&#123;i&#125;_tophat.log</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>运行tophat nohup ./toptat2.sh &amp;</p>
<p>结果回显</p>
<img src="/2019/05/17/RNA-seq/7d932ca52406b48cc9fd31ff7677b7cc.png" class="">

<h2 id="4-2-Cufflinks序列组装"><a href="#4-2-Cufflinks序列组装" class="headerlink" title="4.2 Cufflinks序列组装"></a>4.2 Cufflinks序列组装</h2><p>Cufflinks安装，将文件软连接到环境变量中</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">./configure --prefix=/path/to/cufflinks/install --with-boost=/usr/local/ --with-eigen=/usr/local/include//Eigen/</span><br><span class="line">make</span><br><span class="line">make install</span><br></pre></td></tr></table></figure>

<p>注意，Cufflinks软件安装需要在python2.7条件下使用，因此要切换默认python为2.7</p>
<table>
<thead>
<tr>
<th>cuffdiff -o diff_out_new -b cufflinks/genome.fa -p 8 -L set1CD_24h1,set1CD_24h2,set1CD_48h3,set1CD_48h4,set1CD_7.5h5,set1CD_7.5h6,set1CD_7.5h7,set1CD_0h8,set1CD_0h9,spp1_24h10,spp1_24h11,spp1_48h12,spp1_48h13,spp1_48h14,spp1_48h15,spp1_48h16,spp1_7.5h17,spp1_7.5h18,spp1_7.5h19,spp1_0h20,spp1_0h21,spp1_0h22,spp1_0h23,spp1_0h24,swd1_24h25,swd1_24h26,swd1_24h27,swd1_48h28,swd1_48h29,swd1_48h30,swd1_48h31,swd1_7.5h32,swd1_7.5h33,swd1_7.5h34,swd1_0h35,swd1_0h36,swd1_0h37,swd1_0h38,wt_24h39,wt_24h40,wt_24h41,wt_24h42,wt_24h43,wt_48h44,wt_48h45,wt_48h46,wt_48h47,wt_48h48,wt_7.5h49,wt_7.5h50,wt_7.5h51,wt_7.5h52,wt_7.5h53,wt_0h54,wt_0h55,wt_0h56,wt_0h57,wt_0h58 -u cufflinks/result.gtf ./tophat2_align/new_SRR6350197_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350198_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350199_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350200_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350201_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350202_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350203_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350204_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350205_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350206_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350207_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350208_thout_new/accepted_hits.bam ……（全部样本）</th>
</tr>
</thead>
</table>
<p>大概跑了5个小时后提示segmentation fault(core drump)</p>
<p>猜测是内存不够，只能跑了10个样本</p>
<table>
<thead>
<tr>
<th>cuffdiff -o diff_out -b cufflinks/genome.fa -p 8 -L C1,C2,C3,C4,C5,T1,T2,T3,T4,T5 -u cufflinks/result.gtf ./tophat2_align/new_SRR6350208_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350209_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350210_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350211_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350212_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350216_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350217_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350218_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350219_thout_new/accepted_hits.bam ./tophat2_align/new_SRR6350220_thout_new/accepted_hits.bam</th>
</tr>
</thead>
</table>
<img src="/2019/05/17/RNA-seq/1a735a6460a82868d51db99404e79dde.png" class="">

<p>从生成文档中选择genes.fpkm_tracking文档</p>
<img src="/2019/05/17/RNA-seq/99763c1ae80ed8b750c15f1c1b7fc636.png" class="">

<p>内容回显</p>
<img src="/2019/05/17/RNA-seq/054179835223e55d448b8b0b8e68ad9e.png" class="">

<p>此为TopHat2+cufflinks流程的结果文档</p>
<ol start="5">
<li><h1 id="Hisat2-StringTie-Ballgown流程"><a href="#Hisat2-StringTie-Ballgown流程" class="headerlink" title="Hisat2+StringTie+Ballgown流程"></a>Hisat2+StringTie+Ballgown流程</h1></li>
</ol>
<h2 id="5-1-Hisat2-stringtie"><a href="#5-1-Hisat2-stringtie" class="headerlink" title="5.1 Hisat2+stringtie"></a>5.1 Hisat2+stringtie</h2><p>安装Hisat2、StringTie，将文件软连接到环境变量中</p>
<p>sudo ln -s /mnt/g/RNA_seq/hisat2-2.1.0-Linux_x86_64/hisat2-2.1.0/hisat2<br>/usr/local/bin/hisat2</p>
<p>sudo ln -s /mnt/g/RNA_seq/stringtie-1.3.5/stringtie /usr/local/bin/stringtie</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">for i in `seq 197 254`</span><br><span class="line">do   </span><br><span class="line">hisat2 -p 4 -x /mnt/g/RNA_seq/sra/hisat2_align/r64_tran/genome_tran -U /mnt/g/RNA_seq/sra/new_SRR6350$&#123;i&#125;.fastq -S /mnt/g/RNA_seq/sra/hisat2_align/hisat2/hisat2_SRR6350$&#123;i&#125;.sam</span><br><span class="line">cd hisat2_align/hisat2 </span><br><span class="line">samtools view -S hisat2_SRR6350$&#123;i&#125;.sam -b &gt; hisat2_SRR6350$&#123;i&#125;.bam</span><br><span class="line">samtools sort hisat2_SRR6350$&#123;i&#125;.bam hisat2_SRR6350$&#123;i&#125;_sorted</span><br><span class="line">stringtie hisat2_SRR6350$&#123;i&#125;_sorted.bam -p 4 -G /mnt/g/RNA_seq/sra/hisat2_align/stringtie/Saccharomyces_cerevisiae.R64-1-1.86.gtf -o stringtie/new_SRR6350$&#123;i&#125;.gtf</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>运行 nohup ./hisat2.sh &amp;</p>
<h2 id="5-2-Ballgown"><a href="#5-2-Ballgown" class="headerlink" title="5.2 Ballgown"></a>5.2 Ballgown</h2><p>Ballgown是R语言的程序包，在R studio中运行</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line"># devtools::install_github(&#x27;alyssafrazee/RSkittleBrewer&#x27;, force=TRUE)</span><br><span class="line">library(ballgown)</span><br><span class="line">library(RSkittleBrewer)</span><br><span class="line">library(genefilter)</span><br><span class="line">library(dplyr)</span><br><span class="line">library(devtools)</span><br><span class="line">setwd(&quot;G:/RNA_seq/sra&quot;)</span><br><span class="line">pheno_data &lt;- read.table(&quot;ballgown/phenodata.txt&quot;,header = T)</span><br><span class="line">bg = ballgown(dataDir = &quot;ballgown&quot;, samplePattern = &quot;SRR&quot;, pData=pheno_data)</span><br><span class="line">#滤掉了样本间差异少于一个转录本的数据</span><br><span class="line">bg_filt = subset(bg,&quot;rowVars(texpr(bg)) &gt;1&quot;,genomesubset=TRUE)  </span><br><span class="line">result_transcripts=stattest(bg_filt,feature = &quot;transcript&quot;,covariate = &quot;celllines&quot;,adjustvars = c(&quot;time&quot;), getFC=TRUE,meas=&quot;FPKM&quot;)</span><br><span class="line">result_genes=stattest(bg_filt,feature = &quot;gene&quot;,covariate = &quot;celllines&quot;,adjustvars = c(&quot;time&quot;), getFC=TRUE,meas=&quot;FPKM&quot;)</span><br><span class="line">result_transcripts=data.frame(geneNames=ballgown::geneNames(bg_filt),geneIDs=ballgown::geneIDs(bg_filt),result_transcripts)</span><br><span class="line">result_transcripts=arrange(result_transcripts,pval)</span><br><span class="line">result_genes=arrange(result_genes,pval)</span><br><span class="line">#把结果写入csv文件</span><br><span class="line">write.csv(result_transcripts, &quot;transcript_results.csv&quot;,row.names=FALSE)</span><br><span class="line">write.csv(result_genes, &quot;gene_results.csv&quot;,row.names=FALSE)</span><br><span class="line">data &lt;- subset(result_transcripts,result_transcripts$qval&lt;0.001)</span><br><span class="line">gene &lt;- subset(result_genes,result_genes$qval&lt;0.001)</span><br><span class="line">dim(gene)</span><br><span class="line"># 提取FPKM值</span><br><span class="line">#方便作图将其log转换fpkm = texpr(bg_filt,meas=&quot;FPKM&quot;)</span><br><span class="line">fpkm = log2(fpkm+1)</span><br><span class="line">write.table(fpkm,&quot;fpkm.txt&quot;,quote=FALSE)</span><br></pre></td></tr></table></figure>


<p><strong>查看各样本的表达情况</strong></p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">#设置颜色</span><br><span class="line">tropical= c(&#x27;darkorange&#x27;, &#x27;dodgerblue&#x27;,&#x27;hotpink&#x27;, &#x27;limegreen&#x27;, &#x27;yellow&#x27;)</span><br><span class="line">palette(tropical)</span><br><span class="line">setwd(&quot;G:/RNA_seq/sra/ballgown&quot;)</span><br><span class="line">pdf(&#x27;boxplot.pdf&#x27;,width=10, height=10)</span><br><span class="line">boxplot(fpkm,col=as.numeric(pheno_data$time),las=2,ylab=&#x27;log2(FPKM+1)&#x27;)</span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>

<img src="/2019/05/17/RNA-seq/dff29a695d77b5185e8bf605f1dffe08.png" class="">

<p>图 7 各样本表达情况</p>
<p>就结果而言，各样本除224外的表达值相差不大</p>
<p><strong>就单个转录本的查看在样品中的分布</strong></p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">ballgown::transcriptNames(bg_filt)[10]</span><br><span class="line">ballgown::geneNames(bg_filt)[10]</span><br><span class="line">str = paste(ballgown::geneNames(bg_filt)[10],&#x27; : &#x27;,ballgown::transcriptNames(bg_filt)[10])</span><br><span class="line">str</span><br><span class="line"># TDA8:YAL064C-A</span><br><span class="line">pdf(&#x27;fpkm_boxplot.pdf&#x27;,width=10, height=10)</span><br><span class="line">boxplot(fpkm[12,] ~ pheno_data$celllines, border=c(1,2), main=&quot;TDA8:YAL064C-A&quot;, pch=19, xlab=&quot;celllines&quot;, ylab=&#x27;log2(FPKM+1)&#x27;)</span><br><span class="line">points(fpkm[12,] ~ jitter(as.numeric(pheno_data$celllines)),col=as.numeric(pheno_data$celllines))</span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>

<img src="/2019/05/17/RNA-seq/aea5a78643cef04c7f00e28b0f397adf.png" class="">

<p>图 8 TDA8在样品中的分布</p>
<p><strong>剪切方式分析</strong></p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">pdf(&#x27;plotTranscripts.pdf&#x27;,width=10, height=10)</span><br><span class="line">plotTranscripts(ballgown::geneIDs(bg_filt)[1679], bg_filt, main=c(&#x27;Gene MSTRG.1517.3 in sample SRR6350197&#x27;), sample=c(&#x27; SRR6350197&#x27;))</span><br><span class="line">dev.off()</span><br><span class="line">#以处理方式为区分查看基因表达情况</span><br><span class="line">png(&#x27;plotMeans.png&#x27;)</span><br><span class="line">plotMeans(&#x27;MSTRG.1517&#x27;, bg_filt,groupvar=&quot;celllines&quot;,legend=FALSE)</span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>

<img src="/2019/05/17/RNA-seq/294c434dbe251f4b64dc989235cb6b9a.png" class="">

<p>图 9 特定基因在特定样本中的表达情况</p>
<ol start="6">
<li><h1 id="差异表达-聚类分析"><a href="#差异表达-聚类分析" class="headerlink" title="差异表达+聚类分析"></a>差异表达+聚类分析</h1></li>
</ol>
<h2 id="6-1-TopHat2-Cufflinks流程"><a href="#6-1-TopHat2-Cufflinks流程" class="headerlink" title="6.1 TopHat2 + Cufflinks流程"></a>6.1 TopHat2 + Cufflinks流程</h2><figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">setwd(&quot;G:/RNA_seq/sra/diff_out&quot;)</span><br><span class="line">gene&lt;-read.table(&quot;using_fpkm.txt&quot;,header=T)</span><br><span class="line">gene0&lt;-gene</span><br><span class="line">row.names(gene)&lt;-gene[,1]</span><br><span class="line">gene&lt;-gene[,-1]</span><br><span class="line"></span><br><span class="line">p_value&lt;-c()</span><br><span class="line">t.test()</span><br><span class="line">spp4&lt;-gene[,c(1,2,3,4,5)]</span><br><span class="line">spp0&lt;-gene[,c(6,7,8,9,10)]</span><br><span class="line"></span><br><span class="line">for(i in 1:nrow(gene))&#123;</span><br><span class="line">    p_value[i]&lt;-t.test(spp4[i,],spp0[i,])$p.value</span><br><span class="line">&#125;</span><br><span class="line">gene$p_value&lt;-p_value</span><br><span class="line">gene_p_005 &lt;- gene[p_value&lt;0.05,]</span><br><span class="line"></span><br><span class="line">gene_p_001 &lt;- gene[p_value&lt;0.01,]</span><br><span class="line"></span><br><span class="line">#倍数法</span><br><span class="line">spp4_avg&lt;-apply(spp4,1,mean)</span><br><span class="line">spp0_avg&lt;-apply(spp0,1,mean)</span><br><span class="line">gene$spp4_avg&lt;-spp4_avg</span><br><span class="line">gene$spp0_avg&lt;-spp0_avg</span><br><span class="line">gene_fold_2 &lt;- gene[2^(log2(spp4_avg/spp0_avg))&gt;2,]</span><br><span class="line"></span><br><span class="line">gene_fold_3 &lt;- gene[2^(log2(spp4_avg/spp0_avg))&gt;3,]</span><br><span class="line"></span><br><span class="line">#筛选</span><br><span class="line">#t 0.01  倍数 2</span><br><span class="line">gene_t005_fold2&lt;-gene_fold_2[gene_fold_2$p_value&lt;0.05,]</span><br><span class="line"></span><br><span class="line">gene$avg_cha&lt;-abs(spp4_avg-spp0_avg)</span><br><span class="line">gene_z&lt;-vector(length=nrow(gene))</span><br><span class="line">for(i in 1:nrow(gene))&#123;</span><br><span class="line">    gene_z[i]&lt;-(gene$avg_cha[i]-mean(gene$avg_cha))/sd(gene$avg_cha)</span><br><span class="line">&#125;</span><br><span class="line">gene$gene_z&lt;-abs(gene_z)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">gene_zscore_196&lt;-gene[gene$gene_z&gt;1.96,]</span><br><span class="line"># gene_t001_z005&lt;-gene[gene_fold_2$gene_z&gt;1.96,]</span><br><span class="line">gene_zscore_258&lt;-gene[gene$gene_z&gt;2.58,]</span><br><span class="line"></span><br><span class="line">gene_t001_z001&lt;-gene_zscore_258[gene_zscore_258$p_value&lt;0.01,]</span><br><span class="line"></span><br><span class="line">gene_t001_fold2_z005&lt;-gene_t001_fold2[gene_t001_fold2$gene_z&gt;1.96,]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">gene_chayi&lt;-gene_t001_z001[,-11]</span><br><span class="line">gene_chayi&lt;-gene_chayi[,-11]</span><br><span class="line">gene_chayi&lt;-gene_chayi[,-11]</span><br><span class="line">gene_chayi&lt;-gene_chayi[,-11]</span><br><span class="line">gene_chayi&lt;-gene_chayi[,-11]</span><br><span class="line">treats&lt;-c(&quot;48h&quot;,&quot;48h&quot;,&quot;48h&quot;,&quot;48h&quot;,&quot;48h&quot;,&quot;0h&quot;,&quot;0h&quot;,&quot;0h&quot;,&quot;0h&quot;,&quot;0h&quot;)</span><br><span class="line">library(ggplot2)</span><br><span class="line">library(pheatmap)</span><br><span class="line">heatmap&lt;-function(data,mainname,path)&#123;</span><br><span class="line">    pdf(path,onefile = FALSE,width=10, height=10)     #保存地址(save PATH)，onefile删除了第一页的空白</span><br><span class="line">    annotation_col = data.frame(Type = factor(treats))</span><br><span class="line">    rownames(annotation_col) = paste(colnames(data))    #构建annotation_col</span><br><span class="line">    #绘制heatmap(Drawing the heatmap)</span><br><span class="line">    pheatmap(data,</span><br><span class="line">             #cluster_col = FALSE,</span><br><span class="line">             scale=&quot;row&quot;,                           #按行聚类(Clustering by row)</span><br><span class="line">             treeheight_row=100,treeheight_col=20,     #去除聚类树(Remove the clustering tree)</span><br><span class="line">             #加颜色(color)</span><br><span class="line">             color=colorRampPalette(c(&#x27;green&#x27;,&#x27;black&#x27;,&#x27;red&#x27;))(50),</span><br><span class="line">             show_rownames=F,       #去行名(Remove rowname)</span><br><span class="line">             annotation_col = annotation_col,       #将轨道赋给heatmap(Assign orbits to heatmap)</span><br><span class="line">             main=mainname,</span><br><span class="line">             border_color=NA</span><br><span class="line">    )</span><br><span class="line">    dev.off()</span><br><span class="line">&#125;</span><br><span class="line">heatmap(data=gene_chayi,mainname=&quot;Heatmap Of t.test p&lt;005 fold3&quot;,path=&quot;E:/生信/大三/转录组学/heatmap_t001_fold2.pdf&quot;)</span><br></pre></td></tr></table></figure>


<img src="/2019/05/17/RNA-seq/561aaac3cac0f592bbb41fab84fa9365.png" class="">

<p>图 10 cuffdiff-heatmap</p>
<h2 id="6-2-Hisat2-StringTie-Ballgown流程"><a href="#6-2-Hisat2-StringTie-Ballgown流程" class="headerlink" title="6.2 Hisat2+StringTie+Ballgown流程"></a>6.2 <a target="_blank" rel="noopener" href="http://www.baidu.com/link?url=Kj0fL3Abu3_-ihzLoyhv4xWozqWa7Z9FbXq88Md0l9k9GhNXtmOidRs7urh6uaSd">Hisat2+StringTie</a>+Ballgown流程</h2><figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">fpkm&lt;-read.table(&quot;fpkm.txt&quot;,header=T)</span><br><span class="line">gene&lt;-cbind(fpkm[,10:24],fpkm[,39:58])</span><br><span class="line">gene&lt;-as.data.frame(gene)</span><br><span class="line"></span><br><span class="line">p_value&lt;-vector(length=nrow(gene))</span><br><span class="line">t.test()</span><br><span class="line">nwt&lt;-gene[,c(1:14)]</span><br><span class="line">wt&lt;-gene[,c(15:34)]</span><br><span class="line">for(i in 1:nrow(gene))&#123;</span><br><span class="line">    p_value[i]&lt;-t.test(nwt[i,],wt[i,])$p.value</span><br><span class="line">&#125;</span><br><span class="line">gene$p_value&lt;-p_value</span><br><span class="line">gene_p_005 &lt;- gene[p_value&lt;0.05,]</span><br><span class="line">gene_p_0001 &lt;- gene[p_value&lt;0.001,]</span><br><span class="line">gene_chayi&lt;-gene_p_0001[,-36]</span><br><span class="line">gene_chayi&lt;-gene_chayi[,-36]</span><br><span class="line">gene_chayi&lt;-gene_chayi[,-36]</span><br><span class="line">gene_chayi&lt;-gene_chayi[,-36]</span><br><span class="line">gene_chayi&lt;-gene_chayi[,-36]</span><br><span class="line">treats&lt;-c(&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,&quot;spp1&quot;,</span><br><span class="line">    &quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;,&quot;wt&quot;)</span><br><span class="line">library(ggplot2)</span><br><span class="line">library(pheatmap)</span><br><span class="line">heatmap&lt;-function(data,mainname,path)&#123;</span><br><span class="line">    pdf(path,onefile = FALSE,width=10, height=10)     #保存地址(save PATH)，onefile删除了第一页的空白</span><br><span class="line">    annotation_col = data.frame(Type = factor(treats))</span><br><span class="line">    rownames(annotation_col) = paste(colnames(data))    #构建annotation_col</span><br><span class="line">    #绘制heatmap(Drawing the heatmap)</span><br><span class="line">    pheatmap(data,</span><br><span class="line">             #cluster_col = FALSE,</span><br><span class="line">             scale=&quot;row&quot;,                           #按行聚类(Clustering by row)</span><br><span class="line">             treeheight_row=100,treeheight_col=20,     #去除聚类树(Remove the clustering tree)</span><br><span class="line">             #加颜色(color)</span><br><span class="line">             color=colorRampPalette(c(&#x27;green&#x27;,&#x27;black&#x27;,&#x27;red&#x27;))(50),</span><br><span class="line">             show_rownames=F,       #去行名(Remove rowname)</span><br><span class="line">             annotation_col = annotation_col,       #将轨道赋给heatmap(Assign orbits to heatmap)</span><br><span class="line">             main=mainname,</span><br><span class="line">             border_color=NA</span><br><span class="line">    )</span><br><span class="line">    dev.off()</span><br><span class="line">&#125;</span><br><span class="line">heatmap(data=gene_chayi,mainname=&quot;Heatmap Of t.test p&lt;0.001&quot;,path=&quot;E:/生信/大三/转录组学/heatmap_ballgown.pdf&quot;)</span><br></pre></td></tr></table></figure>


<img src="/2019/05/17/RNA-seq/626a83b938f528b6c0586a2450d54f43.png" class="">

<p>图 11 Ballgown-heatmap</p>
<p>两组数据都有成功地将各自的差异表达数据类分离开,下面是两组数据所提取的差异表达值</p>
<table>
<thead>
<tr>
<th>#ballgown<br>SFA1<br>YCG1<br>STE14<br>PAD1<br>FDC1<br>KTI11<br>QDR2<br>RPI1<br>KTR7<br>MPH1<br>MF(ALPHA)2<br>ATE1<br>ECT1<br>CPR2<br>IMD2<br>PRP21<br>MOG1<br>STE3<br>CTK1<br>CWP1<br>MUD2<br>RHO4<br>LAS1<br>BNA5<br>MGR3<br>TRS130<br>DSS1<br>PMS1<br>SMC5<br>ATX2<br>MF(ALPHA)1<br>SSU1<br>BTS1<br>VPS16<br>TAF3<br>KAR3<br>MATALPHA1<br>SKI3<br>OPT2<br>ARR1<br>CTR86<br></th>
<th><br>#cuffdiff<br>RPP1A<br>PDC1<br>RPL38<br>fba1<br>RPS15<br>RPL15A<br>YDR524C-B<br>ADH1<br>RPS21A<br>RPS29B<br>TPI1<br>RPS5<br>ENO2<br>RPP2A<br>RPS27B<br>RpS28a<br>RPS25A<br>RPS2<br>RPP2B<br>RPL32<br>HSP12<br>RPS3<br>RPL31A<br>SeD1<br>RPS12<br>RPS20<br>TDH2<br>RPL27A<br>RPS1B<br>RPS26A<br>RPL3<br>RPS9B<br>hyp2<br>RPL6A<br>EFB1<br>RPP1B<br>RPS22A<br>RPL30<br>AHP1<br>RpS10a<br>TDH3<br>RPS17A<br>RPL26B<br>rpl25<br>YDR524W-C<br>SPO24<br>Rps10b<br>RpS14a<br>TMA19<br>RpL34b<br>rps21b<br>RPS13<br>GPM1<br>RPL29<br>RPS26B</th>
</tr>
</thead>
</table>
<p>![<a target="_blank" rel="noopener" href="http://www.biovenn.nl/0.13014500/biovenn.png]">http://www.biovenn.nl/0.13014500/biovenn.png]</a>(21bb4a1cd1aba0239ba4a7d1cbb6f3ee.png %}</p>
<p>图 12 ballgown-cuffdiff</p>
<p>二者之间并没有交集，但这并不能说明两种方法完全不同。首先，二者进行差异表达所采用的方式不同，ballgown是t检验p&lt;0.001，cuffdiff是t检验p&lt;0.01<br>z检验p&lt;0.01;其次，二者都是从几千个基因差异表达到几十个，因此出现基因完全不同也比较正常。</p>
</div></article><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/RNA-seq/">RNA_seq</a></div><nav id="pagination"><div class="prev-post pull-left"><a href="/2019/07/11/drug-designing/"><i class="fa fa-chevron-left">  </i><span>计算机辅助药物设计（CADD）知识点总结</span></a></div><div class="next-post pull-right"><a href="/2019/05/17/ProteomeInformatics/"><span>蛋白质组信息学知识点总结</span><i class="fa fa-chevron-right"></i></a></div></nav></div></div><footer class="footer-bg" style="background-image: url(/sabar.jpg)"><div class="layout" id="footer"><div class="copyright">&copy;2018 - 2021 By 双语之城</div><div class="framework-info"><span>Driven - </span><a target="_blank" rel="noopener" href="http://hexo.io"><span>Hexo</span></a><span class="footer-separator">|</span><span>Theme - </span><a target="_blank" rel="noopener" href="https://github.com/Molunerfinn/hexo-theme-melody"><span>Melody</span></a></div><div class="busuanzi"><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><span id="busuanzi_container_page_pv"><i class="fa fa-file-o"></i><span id="busuanzi_value_page_pv"></span><span></span></span></div></div></footer><i class="fa fa-arrow-up" id="go-up" aria-hidden="true"></i><script src="https://cdn.jsdelivr.net/npm/animejs@latest/anime.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-animate@latest/velocity.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-ui-pack@latest/velocity.ui.min.js"></script><script src="/js/utils.js?version=1.6.1"></script><script src="/js/fancybox.js?version=1.6.1"></script><script src="/js/sidebar.js?version=1.6.1"></script><script src="/js/copy.js?version=1.6.1"></script><script src="/js/fireworks.js?version=1.6.1"></script><script src="/js/transition.js?version=1.6.1"></script><script src="/js/scroll.js?version=1.6.1"></script><script src="/js/head.js?version=1.6.1"></script><script>if(/Android|webOS|iPhone|iPod|BlackBerry/i.test(navigator.userAgent)) {
  $('#nav').addClass('is-mobile')
  $('footer').addClass('is-mobile')
}</script><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"debug":false,"model":{"jsonPath":"/live2dw/assets/hijiki.model.json"},"display":{"position":"right","width":150,"height":300},"mobile":{"show":true},"log":false});</script></body></html>